### PR TITLE
[XERCESC-2222] DFAContentModel::checkUniqueParticleAttribution(): fix memory leak

### PR DESCRIPTION
**Pull request by rouault**
If a memory allocation of conflictTable[] fails, or later in the
function, the array is not freed.
Fixes https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=38533

### CODE REVIEW

**Review by rleigh-codelibre — APPROVED** [review]
The change looks absolutely fine to me, thanks.  I'll wait for the CI to complete before merging.

**Comment by scantor** [comment]
I applied this to the 3.2 branch.

**Pull request merged**
